### PR TITLE
Update and accept reservation, handled unsuccessful mailing

### PR DESCRIPTION
**Pull request by miccu64**

### CODE REVIEW

**Review by bestemic — CHANGES_REQUESTED** [review]
Overall nice. Soma cosmetic changes needed. And you give me topic to think about about a reservation that conflicts with the current state of activity.

File: backend/src/main/java/com/wpz/rbs/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws
                 .csrf()
                 .disable()
                 .authorizeHttpRequests()
-                .requestMatchers("/import/**", "/room/update/**", "/room/*/conflicts")
+                .requestMatchers("/import/**", "/room/update/**", "/room/*/conflicts", "reservation/manage/**")
```
[the remark this answers is not in the document]
Comment (miccu64) [reply backend/src/main/java/com/wpz/rbs/config/SecurityConfiguration.java:40]: In reservation/** we've also got adding and editing reservation. They should be accessible by typical user, so why should I restrict it to admin only?

[the remark this answers is not in the document]
Comment (bestemic) [reply backend/src/main/java/com/wpz/rbs/config/SecurityConfiguration.java:40]: Edit can only user. And about adding you are right. We need to add everything to protection except creating new reservation.

File: backend/src/main/java/com/wpz/rbs/controller/ReservationController.java [hunk 2]
```diff
@@ -30,14 +35,18 @@ public Reservation getReservationById(@PathVariable int id) {
         return reservationService.getById(id);
     }
 
+    @PatchMapping("{id}")
```
[the remark this answers is not in the document]
Comment (miccu64) [reply backend/src/main/java/com/wpz/rbs/controller/ReservationController.java:38]: "A very important requirement is that PUT is idempotent: if you require side-effects when updating a Group (or changing an activation), you should use PATCH. So, when the update results in e.g. sending out a mail, don't use PUT."
Citation taken from SO - I think you are too much strict in code review. PATCH and PUT both are appropriate in this case IMO.

File: backend/src/main/java/com/wpz/rbs/service/ReservationService.java [hunk 3]
```diff
@@ -39,19 +43,84 @@ public List<Reservation> getAll() {
...
+        if (result != null)
+            return result;
+
+        Reservation savedReservation = reservationRepository.save(new Reservation(reservation));
+        emailService.sendMessageToAdmin(savedReservation);
+        return ResponseEntity.ok(savedReservation.getId());
+    }
+
+    @Transactional
+    public ResponseEntity<?> confirmReservation(int id) throws ParseException {
```
[the remark this answers is not in the document]
Comment (miccu64) [reply backend/src/main/java/com/wpz/rbs/service/ReservationService.java:57]: Ok

File: backend/src/main/java/com/wpz/rbs/model/reservation/PutReservation.java [hunk 4]
```diff
@@ -11,7 +11,7 @@
 @Builder
 @AllArgsConstructor
 @NoArgsConstructor
-public class CreateReservation {
+public class PutReservation {
```
[the remark this answers is not in the document]
Comment (miccu64) [reply backend/src/main/java/com/wpz/rbs/model/reservation/PutReservation.java:14]: Okay, better name than is now

File: backend/src/main/java/com/wpz/rbs/controller/ReservationController.java [hunk 5]
```diff
@@ -30,14 +35,18 @@ public Reservation getReservationById(@PathVariable int id) {
...
+
     @GetMapping("room/{roomId}")
     public List<Reservation> getReservationsByRoom(@PathVariable int roomId) {
         return reservationService.getByRoomId(roomId);
     }
 
-    @GetMapping()
-    public List<Reservation> getAllReservations() {
-        return reservationService.getAll();
+    @PutMapping("manage/{id}")
```
[the remark this answers is not in the document]
Comment (miccu64) [reply backend/src/main/java/com/wpz/rbs/controller/ReservationController.java:48]: I believe PUT is okay in this case, but u r right, POST will be even better

File: backend/src/main/java/com/wpz/rbs/service/ReservationService.java [hunk 6]
```diff
@@ -39,19 +43,84 @@ public List<Reservation> getAll() {
...
+
+        ResponseEntity<?> result = checkNewOrEditedReservation(editedReservation);
+        if (result != null)
+            return result;
+
+        reservation = new Reservation(editedReservation);
+        reservation.setId(reservationId);
+        reservationRepository.save(reservation);
+        emailService.sendChangedMessageToUser(reservation);
+
```
[the remark this answers is not in the document]
Comment (miccu64) [reply backend/src/main/java/com/wpz/rbs/service/ReservationService.java:107]: Done

File: backend/src/main/java/com/wpz/rbs/controller/ReservationController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (miccu64) [reply backend/src/main/java/com/wpz/rbs/controller/ReservationController.java:38]: After reading comment that we should accept when updating now it fully makes sense to change it to put - changed

[the remark this answers is not in the document]
Comment (bestemic) [reply backend/src/main/java/com/wpz/rbs/controller/ReservationController.java:38]: Ok, I don't know about PUT and making another actions.

File: backend/src/main/java/com/wpz/rbs/config/SecurityConfiguration.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (miccu64) [reply backend/src/main/java/com/wpz/rbs/config/SecurityConfiguration.java]: Done

**Review by bestemic — APPROVED** [review]
Thanks for changes. Looks now good.

**Pull request merged**